plugin/trino-hive/src/main/java/io/trino/plugin/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -419,6 +419,40 @@ private static TableStatistics getTableStatistics(
         return result.build();
     }
 
+    private static TableStatistics createEmptyTableStatisticsWithPartitionColumnStatistics(
+            Map<String, ColumnHandle> columns,
+            Map<String, Type> columnTypes,
+            List<HivePartition> partitions)
+    {
+        TableStatistics.Builder result = TableStatistics.builder();
+        for (Map.Entry<String, ColumnHandle> column : columns.entrySet()) {
+            HiveColumnHandle columnHandle = (HiveColumnHandle) column.getValue();
+            if (columnHandle.isPartitionKey()) {
+                result.setColumnStatistics(
+                        columnHandle,
+                        createPartitionColumnStatistics(columnHandle, columnTypes.get(column.getKey()), partitions));
+            }
+        }
+        return result.build();
+    }
+
+    private static ColumnStatistics createPartitionColumnStatistics(HiveColumnHandle column, Type type, List<HivePartition> partitions)
```

Comment:
Note that it is possible to create partitions in the metastore without having any actual files within them. This estimation is making an assumption that the partitions are non-empty. Please add a code comment about this assumption, so that it's clear to someone else reading this that it's not a mistake.